packages/zeebe-element-templates-json-schema/src/defs/properties.json

```diff
@@ -224,6 +229,56 @@
             ]
           }
         }
+      },
+      {
+        "if": {
+          "properties": {
+            "type": {
+                "enum": [
+                  "String",
+                  "Text"
+                ]
+            }
+          },
+          "required": [
+            "feel"
+          ]
+        },
+        "then": {
+          "properties": {
+            "feel": {
+              "enum": [
+                "optional",
+                "required"
+              ]
+            }
+          }
+        }
+      },
+      {
+        "if": {
+          "properties": {
+            "type": {
+                "enum": [
+                  "Boolean",
+                  "Number"
+                ]
+            }
+          },
+          "required": [
+            "feel"
+          ]
+        },
+        "then": {
+          "properties": {
+            "feel": {
+              "enum": [
+                "optional",
+                "staticValue"
```

Comment:
Why can't this be `"feel": "required"`?

Reply:
I guess it's because the number/boolean controls wouldn't be displayed, but in that case the template developer may want to express the semantics of the field: "This has to be an expression which returns boolean" for which `{ type: "Boolean", feel: "required" }` makes sense.

Reply:
My initial reasoning was that `"feel": "required"` will always result in just a FEEL editor without further validation. It does no longer offer the Restrictions/UI of Number or Boolean fields. Therefore I think we should disallow it if the result we display would be a Text Input anyway.

You make a valid point and I see how this can offer more options, so I'll add adjust it

Reply:
Did we consider `static` as a simpler version of `staticValue`? I'd imagine both would need explanation anyway, and in these cases I'd prefer the shorter form.